skyportal/tests/api/test_annotations.py

```diff
@@ -397,3 +397,29 @@ def test_post_invalid_data(annotation_token, public_source, public_group):
 
     assert status == 400
     assert 'Invalid data' in data["message"]
+
+
+def test_fetch_all_annotations_on_obj(annotation_token, public_source, public_group):
+    status, data = api(
+        'POST',
+        f'sources/{public_source.id}/annotations',
+        data={
+            'obj_id': public_source.id,
+            'origin': 'kowalski',
+            'data': {'offset_from_host_galaxy': 1.5},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+    assert status == 200
+
+    status, data = api(
+        'GET',
+        f'sources/{public_source.id}/annotations',
+        token=annotation_token,
+    )
+
+    assert status == 200
+    assert len(data['data']) == 1
```

Comment:
We should probably check with more than one annotation, to ensure that this follows the "fetch all" path.

Reply:
This already checks that - the data would be an object, not a length one list, otherwise

Reply:
Two would make that more obvious, but no biggie.

Reply:
I'm not sure I understand. If we posted two annotations with different origins, we would get back two annotations...

Reply:
To be extra clear: this is already testing that we're going through the multiple annotations path, as we're ensuring that the response is a length-one list and that the first element is the actual annotation data we expect

Reply:
What's your confusion, Guy?